plugins/orchestrator/src/components/WorkflownstancesViewerPage/ProcessTimeline.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useCallback, useMemo } from 'react';
+import Moment from 'react-moment';
+
+import { InfoCard } from '@backstage/core-components';
+
+import { Tooltip } from '@material-ui/core';
+import CancelOutlinedIcon from '@material-ui/icons/CancelOutlined';
+import CheckCircleIcon from '@material-ui/icons/CheckCircle';
+import OfflineBoltOutlinedIcon from '@material-ui/icons/OfflineBoltOutlined';
+
+import {
+  NodeInstance,
+  ProcessInstance,
+  ProcessInstanceError,
+} from '@janus-idp/backstage-plugin-orchestrator-common';
+
+import { Paragraph } from '../Paragraph/Paragraph';
+
+interface ProcessTimelineProps {
+  selectedInstance: ProcessInstance | undefined;
+}
+
+export const ProcessTimeline = (props: ProcessTimelineProps) => {
+  const { selectedInstance } = props;
+
+  const nodes: NodeInstance[] = useMemo(
+    () => selectedInstance?.nodes ?? [],
+    [selectedInstance],
+  );
+  const error: ProcessInstanceError | undefined = useMemo(
+    () => selectedInstance?.error ?? undefined,
+    [selectedInstance],
+  );
+
+  const render = (icon: JSX.Element, node: NodeInstance) => {
```

Comment:
This can be a separate React component

Reply:
There is a refactoring going on and this component will likely be replaced with new code.